src/ray/object_manager/object_directory.cc

```diff
@@ -0,0 +1,112 @@
+#include <iostream>
+
+#include "object_directory.h"
+
+using namespace std;
+
+namespace ray {
+
+ObjectDirectory::ObjectDirectory() = default;
+ObjectDirectory::~ObjectDirectory() = default;
```

Comment:
Remove these two lines.  IIRC you don't need these if you want to keep default ctor and dtor.  Bottomline, put this in the header.  (Reason being user of this class will glance header first, and this piece of information should be conveyed in the header.)

Reply:
Ah, I could've just pointed you to the (Google) style guide: "Every class's public interface should make explicit which copy and move operations the class supports. This should usually take the form of explicitly declaring and/or deleting the appropriate operations in the public section of the declaration." https://google.github.io/styleguide/cppguide.html#Copyable_Movable_Types